rtos/Thread.cpp

```diff
@@ -334,6 +334,43 @@ osStatus Thread::wait(uint32_t millisec) {
     return osDelay(millisec);
 }
 
+osStatus Thread::wait_until(uint64_t millisec) {
```

Comment:
One possible alternative, is rather than converting up to 64 bits, you could instead convert down to 32 bits. You could still handle the case there the time is slightly in the past inside wait_until. The code to perform the delay would also be simplified substantially.

This would make the behavior for things like Semaphores, Mutexes and Events implementing wait_until more consistent, since it would just be a single call, rather than a loop which will break any ordering RTX tries to keep.